sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/management/ManagementClientAsync.java

```diff
@@ -86,14 +78,93 @@ public ManagementClientAsync(ConnectionStringBuilder connectionStringBuilder) {
      * @param namespaceEndpointURI - URI of the namespace connecting to.
      * @param clientSettings - client settings.
      */
-    public ManagementClientAsync(URI namespaceEndpointURI, ClientSettings clientSettings) {
+//    public ManagementClientAsync(URI namespaceEndpointURI, ClientSettings clientSettings) {
+//        this.namespaceEndpointURI = namespaceEndpointURI;
+//        this.clientSettings = clientSettings;
+//        DefaultAsyncHttpClientConfig.Builder clientBuilder = Dsl.config()
+//                .setConnectTimeout((int) CONNECTION_TIMEOUT.toMillis())
+//                .setRequestTimeout((int) this.clientSettings.getOperationTimeout().toMillis());
+//        this.asyncHttpClient = asyncHttpClient(clientBuilder);
+//    }
+
+    public ManagementClientAsync(URI namespaceEndpointURI, ClientSettings clientSettings, DefaultAsyncHttpClientConfig.Builder httpClientBuilder) {
+        this.namespaceEndpointURI = namespaceEndpointURI;
```

Comment:
Should we consider any null checks to avoid inadvertent NPEs? (ie. Objects.requireNotNull)

Reply:
Same with the other public constructors.